spec/dynamoid/criteria/chain_spec.rb

```diff
@@ -769,6 +769,27 @@ def request_params
     end
   end
 
+
+  context 'single table inheritance' do
+    describe 'where' do
+      it 'honors STI' do
+        Vehicle.create(description: 'Description')
+        car = Car.create(description: 'Description')
+
+        expect(Car.where(description: 'Description').all).to eq [car]
+      end
+    end
+
+    describe 'all' do
+      it 'honors STI' do
+        Vehicle.create(description: 'Description')
+        car = Car.create
+
+        expect(Car.all).to eq [car]
+      end
+    end
```

Comment:
Also add the test where find `Car.all` and make sure it is including all subclass objects so create a `Car` and `Cadillac`.

Reply:
@richardhsu 
I am not sure
We already have such test case here https://github.com/Dynamoid/Dynamoid/pull/185/files#diff-686b0e45ce18ac9c53508980fafe31a0R554

TBH I dislike how test cases are organized and grouped with contexts and files in this project. It's difficult for me to find the right place for new test and to find a place where some functionality is already testing

Reply:
Yeah, I agree, it's difficult. And thanks, I missed that test, I was searching through them to see and guess I just happened to miss that one! Looks good then!